File: java/Item.java
import java.util.*;

public abstract class Item{

	/* Atributos */

	private String name;
	private double price; 

	/* Construtor */

	public Item(String name, double price){

		// Verificando os intervalos, caso esteja fora atribui-se o menor valor
		if (price < 1)
			price = 1;		

		this.name = name;
		this.price = price;
	}

	public Item clone() {
		return (Item)this.clone();
	}

	/* Métodos Getters */

	public String getName(){
		return name;
	}

	public double getPrice(){
		return price;
	}

	/* Métodos */

	// Equipa um 'item'(implementado em Armor e Weapon, itens que podem ser equipados)
	public abstract void equip(Character character);

	// Desequipa um 'item'(implementado em Armor e Weapon, itens que podem ser equipados)
	public abstract void unequip(Character character);

	// Retorna valor de ataque de 'character'(implementado em Armor e Weapon)
	public abstract int getAttackPts();

	// Retorna valor de defesa de 'character'(implementado em Armor e Weapon)
	public abstract int getDefensePts();

	// Retorna o valor de 'weight' do item 'armor'(implementado em Armor)
	public double getWeight(){ return 0.0; }	

}

File: cc/src/Item.cpp
#include <Item.h>

Item::Item(string name, double price){

	/* verificando os intervalos, caso esteja fora atribui-se o menor valor */
	if (price < 1)
		price = 1;		

	this->name = name;
	this->price = price;
}

Item::Item(const Item& item){ 
	this->name = item.name;
	this->price = item.price;	

}
Item::~Item(){ }

string Item::getName(){
	return name;
}

double Item::getPrice(){
	return price;
}

// explicacao do porque de ter criado esses metodos e nao aproveitado o use estao no Item.h
void Item::equip(Character* character){ }
void Item::unequip(Character* character){ }

// mandei retornar 0 para tirar os warning
int Item::getAttackPts(){return 0;}
int Item::getDefensePts(){return 0;}
double Item::getWeight(){return 0.0;}


